src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceIndexingOperationListenerImpl.java

```diff
@@ -67,7 +67,8 @@ public GetResult getGetResult() {
 
     @Override
     public void postDelete(final ShardId shardId, final Delete delete, final DeleteResult result) {
-        if (auditlog.getComplianceConfig().isEnabled()) {
+        final ComplianceConfig complianceConfig = auditlog.getComplianceConfig();
+        if (complianceConfig != null && complianceConfig.writeHistoryEnabledForIndex(shardId.getIndexName())) {
```

Comment:
Should we move this to separate method, similar to "isLoggingWriteDiffEnabled" ?